xapian-core/cluster/cluster.cc

```diff
@@ -132,92 +141,357 @@ TermListGroup::get_termfreq(const string &tname) {
 }
 
 doccount
-ClusterSet::num_of_clusters() {
-    LOGCALL(API, doccount, "ClusterSet::num_of_clusters()", "");
+ClusterSet::size() {
+    LOGCALL(API, doccount, "ClusterSet::size()", NO_ARGS);
     return clusters.size();
 }
 
 doccount
 ClusterSet::cluster_size(clusterid cid) {
     LOGCALL(API, doccount, "ClusterSet::cluster_size()", cid);
-    map<clusterid, vector<Document> >::iterator i;
-    i = clusters.find(cid);
-    if (i == clusters.end())
+    unsigned int s = clusters.size();
+    if (cid >= s)
 	throw RangeError("The mentioned clusterid was out of range", 103);
```

Comment:
Why are you passing `103` here?  If I read the overloads right, it'll be interpreted as an `errno` value, but what a literal `103` means depends on the platform (and for Linux, it even varies between architectures in some cases).  For me, `103` is `ECONNABORTED` - "Software caused connection abort", which seems to make no sense.

I don't think you want to be injecting an `errno` value for a `RangeError` anyway, but more generally if you do want to refer to an error value, use the `EXXX` constant for it.